compiler/daml-lf-tools/src/DA/Daml/LF/Simplifier.hs

```diff
@@ -356,6 +357,96 @@ simplifyExpr world = fst . cata go
       -- e    ==>    e
       e -> (embed (fmap fst e), infoStep world (fmap snd e))
 
+-- Top-down inliner for simple functions.
+inlineExpr :: World -> Expr -> Expr
+inlineExpr _world = go
+  where
+    mkApp :: Expr -> Expr -> Expr
+    mkApp (ELocation l e1) e2 = ELocation l (ETmApp e1 e2)
+    mkApp e1 e2 = ETmApp e1 e2
+
+    -- Special cases. We split the general cases into a separate function
+    -- 'goExpr' so that GHC's pattern exhaustiveness checker works better.
+    go :: Expr -> Expr
+    go = \case
+        EApps (EVal x) [TyArg _, TyArg _, TyArg _, TmArg f, TmArg g, TmArg h]
+            | qualModule x == ModuleName ["GHC", "Base"]
```

Comment:
Do we have a way to further restrict this to the right package id as well? If not, that's probably not a problem for now since I don't expect anybody to name their module `GHC.Base`.